### PR TITLE
added keypoints extraction function

### PR DESCRIPTION
**Pull request by wuyenlin**
Hi, I have added keypoints extraction feature, which can be triggered by simply adding `extract_kpts` in the command line.
The keypoints per frame will be stored in a .npz file in the repository root folder.

### CODE REVIEW

**Comment by stefanopini** [comment]
Hi,
Thank you very much!
I'll review the code and let you know if there's anything to update before merging 🙂

File: scripts/live-demo.py [hunk 1]
```diff
@@ -160,5 +169,6 @@ def main(camera_id, filename, hrnet_c, hrnet_j, hrnet_weights, hrnet_joints_set,
                                          "set to `cuda:IDS` to use one or more specific GPUs "
                                          "(e.g. `cuda:0` `cuda:1,2`); "
                                          "set to `cpu` to run on cpu.", type=str, default=None)
+    parser.add_argument("--extract_kpts", help="save output keypoints", action="store_true")
```
Comment (stefanopini) [inline scripts/live-demo.py]:
```suggestion
    parser.add_argument("--extract_kpts", help="save output keypoints in numpy format", action="store_true")
```

File: scripts/live-demo.py [hunk 2]
```diff
@@ -103,7 +106,12 @@ def main(camera_id, filename, hrnet_c, hrnet_j, hrnet_weights, hrnet_joints_set,
             frame = draw_points_and_skeleton(frame, pt, joints_dict()[hrnet_joints_set]['skeleton'], person_index=pid,
                                              points_color_palette='gist_rainbow', skeleton_color_palette='jet',
                                              points_palette_samples=10)
+            if extract_kpts:
+                dic[frame_count] = pt[:, :2]
+            # print("\nframe: {}".format(frame_count))
 
+        if extract_kpts:
+            np.savez("output_kpts", dic)
```
Comment (stefanopini) [inline scripts/live-demo.py]: This should be moved outside the while loop. Here it slows down the performance and does a lot of unnecessary writes.

File: scripts/live-demo.py [hunk 3]
```diff
@@ -15,17 +15,18 @@
...
+        print("Using GPU")
     else:
         if torch.cuda.is_available():
             torch.backends.cudnn.deterministic = True
             device = torch.device('cuda')
         else:
             device = torch.device('cpu')
+            print("Using CPU")
 
-    # print(device)
```
Comment (stefanopini) [inline scripts/live-demo.py]: Please keep this line and remove the two prints above

File: scripts/live-demo.py (hunk 2 above)
Comment (stefanopini) [reply scripts/live-demo.py]: You can also consider to use savez_compressed in place of savez

File: scripts/live-demo.py [hunk 4]
```diff
@@ -59,6 +60,8 @@ def main(camera_id, filename, hrnet_c, hrnet_j, hrnet_weights, hrnet_joints_set,
         prev_person_ids = None
         next_person_id = 0
 
+    frame_count = 0
+    dic = {}
```
Comment (stefanopini) [inline scripts/live-demo.py]: Use meaningful names, e.g. `pts_dict`

**Review by stefanopini — CHANGES_REQUESTED** [review]
Thank you for your pull request.
Please correct these few things and then I'll proceed merging!

**Comment by wuyenlin** [comment]
Hi, thanks for taking the time to review!
I have fixed the code accordingly, please let me know if there is any further issues.

**Review by stefanopini — APPROVED** [review]

**Comment by stefanopini** [comment]
Wow, thank you!
Everything looks good to me, I'll merge the changes into the master branch 🙂

**Pull request merged**